libraries/AC_Fence/AC_Fence.cpp

```diff
@@ -326,12 +337,37 @@ bool AC_Fence::pre_arm_check(const char* &fail_msg) const
         return false;
     }
 
-    if (_alt_max < _alt_min) {
+#if AP_TERRAIN_AVAILABLE
+    if (!Location::valid_alt_frame(_alt_type)) {
+        fail_msg = "invalid FENCE_ALT_TYPE";
+        return false;
+    }
```

Comment:
```suggestion
    if (!Location::valid_alt_frame(_alt_type)) {
        fail_msg = "invalid FENCE_ALT_TYPE";
        return false;
    }
#if AP_TERRAIN_AVAILABLE
```

we use  `_alt_type` regardless of whether terrain is enabled